### PR TITLE
VACMS-5985 alert block governance

### LINKED ISSUE #5985: Improve governance of Alert blocks and Alert Scope field
ACs
- [x] Only content admins can create new alert blocks - this removes UI cruft 
- [x] As non-content-admin, I cannot edit or access the Scope field on any alert block. (currently a major content governance loophole)
- [x] Banner or in-page alert? is a required field and defaults to in-page Alert (admin/structure/block/block-content/manage/alert/fields/block_content.alert.field_is_this_a_header_alert_)
- [x] The Scope field is only visible if "Banner alert" has been selected in field_is_this_a_header_alert_ 
- [ ] That scope field can only be edited through fulfillment request via CMS help desk

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
Closes #5985

## Testing done
Visual

## QA steps
**Scenario 1**
 - As user 1282 on `block/add/alert` confirm that you can create a block.
 - [x] Confirm that you can access the scope field
 - [x] Confirm Banner or in-page alert? is a required field and defaults to in-page Alert
 - [x] Confirm the Scope field is only visible if "Banner alert" has been selected in field_is_this_a_header_alert_
 

- - -
**Scenario 2**
 - As user 1282 on `block/95` confirm that you can access the scope field
 - [x] Confirm Banner or in-page alert? is a required field and defaults to in-page Alert
 - [x] Confirm the Scope field is only visible if "Banner alert" has been selected in field_is_this_a_header_alert_
 - [x] Set Section to VA West Palm Beach health care and save block
- - -
**Scenario 3**
 - As user 2219 on `block/add/alert` 
 -  [x] confirm that access denied message appears
- - -
**Scenario 4**
 - As user 2219 on `block/95` 
 - [x] confirm that you cannot edit or access the Scope field
 - [x] Confirm Banner or in-page alert? is a required field is set  to `in-page Alert`
 - [x] confirm his setting can not be changed.
 - [x] Confirm the Scope field is not available and can not be edited.

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -23914,5 +23914,5 @@
     "platform-overrides": {
         "php": "7.3"
     },
-    "plugin-api-version": "2.1.0"
+    "plugin-api-version": "2.0.0"
 }
```
Comment (swirtSJW) [inline composer.lock:23918]: I think this should not be included.

File: phpstan-baseline.neon [hunk 2]
```diff
@@ -1,5 +1,10 @@
 parameters:
 	ignoreErrors:
+		-
+			message: "#^Call to an undefined method Drupal\\\\Core\\\\Form\\\\FormInterface\\:\\:getEntity\\(\\)\\.$#"
+			count: 2
+			path: docroot/modules/custom/disable_node_menu_item/disable_node_menu_item.module
```
Comment (swirtSJW) [inline phpstan-baseline.neon:6]: This module is no longer in  our codebase.  Do we really need this ignore?

**Review by swirtSJW — CHANGES_REQUESTED** [review]
A couple odd things  (see comments below)

**Comment by ethanteague** [comment]
Apologies for sleeping on this one, @swirtSJW - you reviewed this several days ago, and I forgot to circle back.

**Review by swirtSJW — CHANGES_REQUESTED** [review]
I'm getting a fail on the first step As Lisa (content admin) I do not have access to the 
https://pr6045-hgzahimmyyzt89bvkbxbdftigxnho3vr.ci.cms.va.gov/block/add/alert
image

Can you make the QA steps a little more specific, so I know what you tested already and make it a bit faster for me to test.

**Comment by ethanteague** [comment]
Updated, @swirtSJW :+1:

**Review by swirtSJW — CHANGES_REQUESTED** [review]
I got a fail on the scenerio 1. Scope is visible on node add by default.


I am also not really following why `field-is-this-a-header-alert` is required when there are only two options and one is set by default.  It is impossible to unset this field.

**Comment by ethanteague** [comment]
> I got a fail on the scenerio 1. Scope is visible on node add by default.

Sorry about that, @swirtSJW  - fixed.

> I am also not really following why `field-is-this-a-header-alert` is required when there are only two options and one is set by default. It is impossible to unset this field.

@kevwalsh set this as a requirement in the task: ```Banner or in-page alert? is a required field and defaults to in-page Alert (admin/structure/block/block-content/manage/alert/fields/block_content.alert.field_is_this_a_header_alert_)```

**Review by swirtSJW — APPROVED** [review]
This all passes QA.  Thanks @ethanteague 

This part of the UI should probably also be hidden it is not a banner alert
image

But lets leave it as is for now, since the next iteration will just be to remove all those fields anyway.

**Comment by swirtSJW** [comment]
THis is ready to go, just waiting on tests to complete.

**Comment by swirtSJW** [comment]
image

**Pull request merged**